pkg/identityserver/gormstore/gateway_store.go

```diff
@@ -213,6 +213,20 @@ func (s *gatewayStore) DeleteGateway(ctx context.Context, id *ttnpb.GatewayIdent
 
 func (s *gatewayStore) RestoreGateway(ctx context.Context, id *ttnpb.GatewayIdentifiers) error {
 	defer trace.StartRegion(ctx, "restore gateway").End()
+	if len(id.Eui) > 0 {
+		query := s.query(ctx, Gateway{}, withGatewayEUI(EUI64(types.MustEUI64(id.Eui).OrZero())))
+		query = selectGatewayFields(ctx, query, []string{"ids.gateway_id", "ids.eui"})
+		var gtwModel Gateway
+		if err := query.First(&gtwModel).Error; err != nil {
+			if gorm.IsRecordNotFoundError(err) {
+				return s.restoreEntity(ctx, id)
+			}
+			return err
+		}
+		if gtwModel.GatewayID != id.GatewayId {
```

Comment:
Is this the right thing to compare ? What if they are from different tenants with the same name ?

Reply:
True, ended up removing this and doing an `Unscoped` search for any gateway that contains the EUI. If more than one if found then the restored gateway will have its own EUI cleaned.